### PR TITLE
feat: add send email Hook

### PR DESCRIPTION
**Pull request by J0**
## What kind of change does this PR introduce?

The send email hook serves as a substitute for the default email client (e.g. GoMail) across all endpoints. 

Supercedes #1496  as it was simpler to visualize the PR when starting from scratch

### CODE REVIEW

**Comment by J0** [comment]
Reverting to draft to do some manual testing  and maybe add a few automated tests

File: internal/api/mail.go [hunk 1]
```diff
@@ -502,3 +488,43 @@ func validateSentWithinFrequencyLimit(sentAt *time.Time, frequency time.Duration
...
+			OTP:             otp,
+			EmailActionType: emailActionType,
+			RedirectTo:      referrerURL,
+			SiteURL:         externalURL.String(),
+			TokenHash:       tokenHash,
+		}
+
+		if emailActionType == mail.EmailChangeVerification && config.Mailer.SecureEmailChangeEnabled && u.GetEmail() != "" {
+			emailData.OTPNew = otpNew
+			emailData.TokenHashNew = u.EmailChangeTokenCurrent
```
Comment (J0) [inline internal/api/mail.go:510]: Either add a comment on why this is so or find a way to fetch this into fn

File: internal/api/mail.go [hunk 2]
```diff
@@ -502,3 +489,48 @@ func validateSentWithinFrequencyLimit(sentAt *time.Time, frequency time.Duration
...
+	case mail.MagicLinkVerification:
+		return mailer.MagicLinkMail(r, u, otp, referrerURL, externalURL)
+	case mail.ReauthenticationVerification:
+		return mailer.ReauthenticateMail(r, u, otp)
+	case mail.RecoveryVerification:
+		return mailer.RecoveryMail(r, u, otp, referrerURL, externalURL)
+	case mail.InviteVerification:
+		return mailer.InviteMail(r, u, otp, referrerURL, externalURL)
+	case mail.EmailChangeVerification:
+		return mailer.EmailChangeMail(r, u, otpNew, otp, referrerURL, externalURL)
```
Comment (J0) [inline internal/api/mail.go:532]: add a default catch all case jic

File: internal/mailer/mailer.go [hunk 3]
```diff
@@ -32,6 +32,16 @@ type EmailParams struct {
 	RedirectTo string
 }
 
+type EmailData struct {
+	OTP             string `json:"otp"`
```
Comment (hf) [inline internal/mailer/mailer.go:36]: Same, should it be `Token` and `TokenNew`?

**Review by hf — APPROVED** [review]

**Pull request merged**